default.nix

```diff
@@ -1,58 +1,44 @@
-with import (fetchTarball "https://github.com/serokell/nixpkgs/archive/master.tar.gz") {
-  config.allowUnfree = true;
-  overlays = [ (import "${fetchGit "ssh://git@github.com:/serokell/serokell-overlay"}/pkgs") ];
-};
+{ pkgs ? import ./nixpkgs.nix }: with pkgs;
 
-with haskell.lib;
+let 
+  stack4nix = fetchGit {
+    url = "https://github.com/serokell/stack4nix";
+    rev = "26a46991270749ecc57d5842d391435b314dc26f";
+  };
+
+  selectAttrs = names: set:
+    lib.genAttrs names (name: lib.getAttr name set);
 
-let
-  getAttrs = attrs: set: lib.genAttrs attrs (name: set.${name});
-  dscp-packages = buildStackApplication rec {
   packages = [
     "disciplina-core" "disciplina-witness" "disciplina-educator"
     "disciplina-wallet" "disciplina-tools" "disciplina-faucet"
   ];
 
-  # Running hpack manually before the build is required
-  # because of the problem in stack2nix -- it builds every
-  # subpackage in a separate environment, thus moving
-  # hpack directory out of the scope.
-  #
-  # See: https://github.com/input-output-hk/stack2nix/pull/107
-
-  src = runCommand "source" { cwd = lib.cleanSource ./.; } ''
-    cp --no-preserve=mode,ownership -r $cwd $out
-
-    for f in $out/{core,witness,educator,tools,wallet,faucet}; do
-      ${haskellPackages.hpack}/bin/hpack $f
-    done
-  '';
-
-  ghc = haskell.compiler.ghc822;
-
-  overrides =
-    final: previous:
-    let overridingSet = (super: with final; {
-          configureFlags = [ "--ghc-option=-Werror" ];
-          doCheck = true;
-          testDepends = [ hspec tasty tasty-discover tasty-hspec ];
-          postInstall = ''
-            mkdir -p $out/share/disciplina/specs $out/etc/disciplina
-            cp ${src}/configuration.yaml $_
-            find ${src}/specs -name '*.yaml' -exec cp '{}' $out/share/disciplina/specs \;
-          '';
-        });
-        overrideModule = prev: overrideCabal prev (overridingSet final);
-    in {
-      rocksdb-haskell = dependCabal previous.rocksdb-haskell [ rocksdb ];
-    } // (lib.mapAttrs (lib.const overrideModule) (getAttrs packages previous));
-  };
+  src = lib.cleanSource ./.;
+
+  overrides = with haskell.lib; final: previous:
+    let
+      defaults = super: with final; {
+        configureFlags = [ "--ghc-option=-Werror" ];
+
+        buildTools = (super.buildTools or []) ++ [ autoexporter tasty-discover ];
+        testDepends = (super.testDepends or []) ++ [ hspec tasty tasty-discover tasty-hspec ];
+
+        doCheck = true;
+        doHaddock = false; # until syntax errors in comments are fixed
+
+        postInstall = ''
+          mkdir -p $out/share/disciplina/specs $out/etc/disciplina
+          cp ${src}/configuration.yaml $_
+          find ${src}/specs -name '*.yaml' -exec cp '{}' $out/share/disciplina/specs \;
```

Comment:
Do we need any of this?

Reply:
I was using it in the CD profiles, since they kept changing the config. But it makes disciplina nix expressions depend on ./. and rebuild every time anything changes, so it's actually terrible. Any alternative besides hardcoding config in CD profiles?

Reply:
I think have a separate derivation (`disciplina-config`) that handles this.

Reply:
Please do that :)